edpm_ansible/ansible_plugins/modules/container_config_hash.py

```diff
@@ -145,6 +139,42 @@ def _update_container_config(self, path, config):
         os.chmod(path, 0o600)
         self.results['changed'] = True
 
+    def _tar_cleanup(self, tarinfo):
+        tarinfo.mtime = 0
+        return tarinfo
+
+    def _create_checksum(self, config_volume):
```

Comment:
We have to pull the logic to add/exclude some files, I guess.

https://github.com/openstack/tripleo-heat-templates/blob/master/common/container-puppet.sh#L164
https://github.com/openstack/tripleo-heat-templates/blob/master/common/container-puppet.sh#L170

Probably we want these override-able so that each role can define own requirements.

Reply:
For what I saw, there's no roles, at least for those that we have imported here, who write their config outside `/var/lib/config-data/ansible-generated` Even `logrotate-crond` and `iscsid` have their folder there. What should we add then?

About exclusions, I implemented the logic inside the `_create_checksum` function, defining an `exclusion` list, now defaulted to `[]`, but it's not referenced in its call yet. 

This module is called by the edpm_container_standalone role without any parameter because it was designed to run on *all* container startup config definition found in `CONTAINER_STARTUP_CONFIG` (which is a constant that points to `/var/lib/edpm-config/container-startup-config`) Handling an exclusion list for each service/role means that:

1. we change the logic behind this entire module, calling it for each service with their own parameters.
2. we add a variable in the container definition json file that enumerate a list of files that need to be excluded from hashing.

Are we sure that we still need to exclude files? Even if those files contains passwords or they binaries (like the ceph keyring linked), should they have the same hash if aren't changed?

I'm waiting for your feedback. (cc @slagle for another opinion)

Reply:
I think option 2 is what I was referring to earlier. I think we could manage the exclusion list in the container json (startup_config_json), and then pass the list into _set_config_hash (and then _create_checksum). So, it would still work per container, without any change to how the module is called.

We likely still need file exclusion for /etc/libvirt/passwd.db at least.

Reply:
@slagle done. I'm searching in the `container_config_json` for an environment var called `EDPM_EXCLUDE_FROM_HASH`. The variable itself is a list of files separated by colon, just like the ${PATH} variable style. e.g. `EDPM_EXCLUDE_FROM_HASH=/etc/nova/nova.conf:/etc/libvirt/passwd.db`